geopandas/sindex.py

```diff
@@ -81,22 +237,137 @@ def is_empty(self):
+
+            if predicate not in self.valid_query_predicates:
+                raise ValueError(
+                    "Got `predicate` = `{}`, `predicate` must be one of {}".format(
+                        predicate, self.valid_query_predicates
+                    )
+                )
+            if isinstance(geometry, geoseries.GeoSeries):
+                geometry = geometry.values.data
+            elif isinstance(geometry, GeometryArray):
+                geometry = geometry.data
+            elif not isinstance(geometry, np.ndarray):
+                geometry = np.asarray(geometry)
+
+            res = super().query_bulk(geometry, predicate)
+
+            if not sort:
+                return res
+            # sort by first array (geometry) and then second (tree)
+            geo_res, tree_res = res
+            indexing = np.lexsort((tree_res, geo_res))
+            return np.vstack((geo_res[indexing], tree_res[indexing]))
+
+        def query(self, geometry, predicate=None, sort=False):
+            """Wrapper for pygeos.query.
+
+            This also ensures a deterministic (sorted) order for the results.
+
+            Parameters
+            ----------
+            geometry : single PyGEOS geometry
+            predicate : {None, 'intersects', 'within', 'contains', 'overlaps', 'crosses', 'touches'}, optional
+                If predicate is provided, a prepared version of the input geometry is tested using
+                the predicate function against each item in the index whose extent intersects the
+                envelope of the input geometry: predicate(geometry, tree_geometry).
+            sort : bool, default False
+                If True, the results will be sorted in ascending order. If False, results are
+                often sorted but there is no guarantee.
+
+            Returns
+            -------
+            matches : ndarray of shape (n_results, )
+                Integer indices for matching geometries from the spatial index.
+            See also
+            --------
+            See PyGEOS.strtree documentation for more information.
+            """  # noqa: E501
+
+            if predicate not in self.valid_query_predicates:
+                raise ValueError(
+                    "Got `predicate` = `{}`; ".format(predicate)
+                    + "`predicate` must be one of {}".format(
+                        self.valid_query_predicates
+                    )
+                )
+
+            if isinstance(geometry, BaseGeometry):
+                # handle shapely geometries
+                if compat.PYGEOS_SHAPELY_COMPAT:
+                    geometry = from_shapely(geometry)
+                # fallback going through WKB
+                elif geometry.is_empty and geometry.geom_type == "Point":
+                    # empty point does not roundtrip through WKB
+                    geometry = from_wkb("POINT EMPTY")
+                else:
+                    geometry = from_wkb(geometry.wkb)
```

Comment:
Since this is duplicating the code in `_shapely_to_geom` / `_shapely_to_pygeos`, it is fine to use those directly instead of duplicating the code. 
(yes, those were coded as "private", but it's fine to use them for now within geopandas)

Reply:
Ok, with that okay I went ahead and imported `_shapely_to_geom` (we don't really need `_shapely_to_pygeos`).